command/agent.go

```diff
@@ -1047,3 +1050,19 @@ func (c *AgentCommand) handleMetrics() http.Handler {
 		}
 	})
 }
+
+func (c *AgentCommand) handleQuit() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Only handle POST/PUT requests
+		switch r.Method {
+		case http.MethodPost:
+		case http.MethodPut:
+		default:
+			w.WriteHeader(405)
+			return
+		}
+
+		c.logger.Debug("received quit request")
```

Comment:
Should we log something about the requesting peer? Could be useful to know who is killing us?

Reply:
I think we need to be careful about logging IPs, as they are personal data according to GDPR regulations. Something to consider in another PR/broader context I think.